query/src/main/scala/filodb/query/exec/rangefn/InstantFunction.scala

```diff
@@ -248,4 +251,22 @@ case class HistogramQuantileImpl(funcParams: Seq[Any]) extends HistToDoubleIFunc
   val q = funcParams(0).asInstanceOf[Number].doubleValue()
 
   final def apply(value: Histogram): Double = value.quantile(q)
+}
+
+/**
+ * Function to extract one bucket from any histogram (could be computed, not just raw).
+ * @param funcParams - a single value which is the Double bucket or "le" to extract.  If it does not correspond
+ *                     to any existing bucket then NaN is returned.
+ */
+case class HistogramBucketImpl(funcParams: Seq[Any]) extends HistToDoubleIFunction {
+  require(funcParams.length == 1, "Bucket/le required for histogram bucket")
+  require(funcParams(0).isInstanceOf[Number], "histogram_bucket parameter must be a number")
+  val bucket = funcParams(0).asInstanceOf[Number].doubleValue()
+
+  final def apply(value: Histogram): Double = {
+    for { b <- 0 until value.numBuckets optimized } {
+      if (Math.abs(value.bucketTop(b) - bucket) <= 0.00001) return value.bucketValue(b)
```

Comment:
Would 0.00001 as threshold be enough ?
If buckets are 1ms, 2ms, 4ms etc and all the bucket limits are in seconds, multiple buckets may match.
10 ^ -10 may be safer.

Reply:
Sure we can use that.

Reply:
@vishramachandran ok changed to 1E-10